### ISSUE TITLE
Encoding fails with "Buffer out of bounds"

### ISSUE THREAD

**Issue by mourner** [mafintosh/protocol-buffers#23]
Test PBF: https://github.com/mapbox/pbf/raw/master/test/fixtures/12665.vector.pbf
Proto: https://raw.githubusercontent.com/mapbox/vector-tile-spec/master/1.0.1/vector_tile.proto

Decoding works perfectly, but trying to roundtrip like this:

```js
messages.Tile.encode(messages.Tile.decode(data));
```

...results in the following error:

```
buffer.js:343
    throw new RangeError('attempt to write beyond buffer bounds');
    ^
RangeError: attempt to write beyond buffer bounds
    at Buffer.write (buffer.js:343:11)
    at Object.exports.string.encode (/Users/mourner/Projects/protocol-buffers/encodings.js:64:12)
    at Object.encode (eval at <anonymous> (/Users/mourner/Projects/protocol-buffers/node_modules/generate-function/index.js:88:21), <anonymous>:8:12)
    at Object.encode (eval at <anonymous> (/Users/mourner/Projects/protocol-buffers/node_modules/generate-function/index.js:88:21), <anonymous>:38:14)
    at Object.encode (eval at <anonymous> (/Users/mourner/Projects/protocol-buffers/node_modules/generate-function/index.js:88:21), <anonymous>:12:14)
    at Object.<anonymous> (/Users/mourner/Projects/pbf/bench/bench.js:16:15)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
```

**Comment by mafintosh** [comment]
This should be fixed by chrisdickinson/varint#12

**Comment by mourner** [comment]
Hmm, I don't think there are any big numbers in the sample data here. Are you sure?

**Comment by mafintosh** [comment]
`layers.*.values.*.int_value` contains some large numbers when i decode your test data. is that incorrect?

**Comment by mafintosh** [comment]
for example:

``` js
...
values:
   [ { string_value: 'park',
       float_value: 0,
       double_value: 0,
       int_value: 0,
       uint_value: 0,
       sint_value: 0,
       bool_value: false },
     { string_value: '',
       float_value: 0,
       double_value: 0,
       int_value: 1000284764947,
       uint_value: 0,
       sint_value: 0,
       bool_value: false },
...
```

**Comment by mafintosh** [comment]
Hmm, for some reason i'm not able to reproduce this anymore (even with the current version varint). I think I reinstalled the dependencies. Could you try installing the latest protocol buffers and check that still is still an issue?

**Comment by mafintosh** [comment]
Nevermind - I can still reproduce this now. This problematic data is actually `layers.*.features.*.id` which contains large integers.

```

  features:
   [ { id: 1000255493390, tags: [Object], type: 1, geometry: [Object] },
     { id: 18446744071093019000
```

**Comment by mourner** [comment]
Wow, you're right. Thanks for looking into this! Waiting for an updated varint package now.

**Comment by mafintosh** [comment]
Latest varint fixes this and is used in 2.4.6. Could you verify that it works?

**Comment by mourner** [comment]
Works perfectly, thanks!
